src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultTagHelperResolver.cs

```diff
@@ -20,68 +19,42 @@ public override TagHelperResolutionResult GetTagHelpers(Compilation compilation,
         {
             var descriptors = new List<TagHelperDescriptor>();
 
-            VisitTagHelpers(compilation, assemblyNameFilters, descriptors);
-            VisitViewComponents(compilation, assemblyNameFilters, descriptors);
-
-            var diagnostics = new List<RazorDiagnostic>();
-            var resolutionResult = new TagHelperResolutionResult(descriptors, diagnostics);
-
-            return resolutionResult;
-        }
-
-        private void VisitTagHelpers(Compilation compilation, IEnumerable<string> assemblyNameFilters, List<TagHelperDescriptor> results)
-        {
-            var types = new List<INamedTypeSymbol>();
-            var visitor = TagHelperTypeVisitor.Create(compilation, types);
-
-            VisitCompilation(visitor, compilation);
+            var providers = new ITagHelperDescriptorProvider[]
+            {
+                new DefaultTagHelperDescriptorProvider(),
+                new ViewComponentTagHelperDescriptorProvider(),
+            };
 
-            var factory = new DefaultTagHelperDescriptorFactory(compilation, DesignTime);
+            var results = new List<TagHelperDescriptor>();
+            var context = new Context(results);
 
-            foreach (var type in types)
+            for (var i = 0; i < providers.Length; i++)
```

Comment:
Doesn't look like you're ordering the providers based on their `Order`. I get why, cause this is only used by tooling but I think consistency here (with the feature) will limit our exposure to bugs. Might make sense to have a provider executor that's used in both locations.